packages/integration-tests/test/actions.spec.ts

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios'
+import * as fs from 'fs';
+import { Config, sleep } from '../../../common'
+import { Watcher } from '@eth-optimism/watcher'
+import { ganache } from '@eth-optimism/ovm-toolchain'
+import { getContractInterface, getContractFactory } from '@eth-optimism/contracts'
+import { JsonRpcProvider, Web3Provider } from '@ethersproject/providers'
+import assert = require('assert')
+
+import {
+  Contract, ContractFactory, providers, Wallet,
+} from 'ethers';
+
+let SimpleStorage
+const L1_USER_PRIVATE_KEY = Config.DeployerPrivateKey()
+const L2_USER_PRIVATE_KEY = Config.DeployerPrivateKey()
+const goerliURL = Config.L1NodeUrlWithPort()
+const optimismURL = Config.L2NodeUrlWithPort()
+const l1Provider = new JsonRpcProvider(goerliURL)
+const l2Provider = new JsonRpcProvider(optimismURL)
+const l1Wallet = new Wallet(L1_USER_PRIVATE_KEY, l1Provider)
+const l2Wallet = new Wallet(L2_USER_PRIVATE_KEY, l2Provider)
+const messengerJSON = getContractInterface('iOVM_BaseCrossDomainMessenger')
+const l2MessengerJSON = getContractFactory('OVM_L2CrossDomainMessenger')
+
+let watcher
+let l1MessengerAddress
+let l2MessengerAddress
+const initWatcher = async () => {
+  const response = await axios.get(`${Config.DeployerUrl()}/addresses.json`)
+  const addresses = response.data
+  l1MessengerAddress = addresses['Proxy__OVM_L1CrossDomainMessenger']
+  l2MessengerAddress = '0x4200000000000000000000000000000000000007' // TODO: pull from deployer once fixed
```

Comment:
Is it possible to pull these from the address resolver?

Reply:
This is outdated now. They get pulled from the Config which gets them from the env vars:

```
L1_MESSENGER_ADDRESS=$(curl --silent $DEPLOYER_HTTP/addresses.json | jq -r .Proxy__OVM_L1CrossDomainMessenger)

exec env \
    ETH1_ADDRESS_RESOLVER_ADDRESS=$ETH1_ADDRESS_RESOLVER_ADDRESS \
    L1_MESSENGER_ADDRESS=$L1_MESSENGER_ADDRESS \
    $cmd
```